source/core/box.ts

```diff
@@ -0,0 +1,225 @@
+            +x, +y, +z,
+
+            +x, -y, +z,
+            +x, -y, -z,
+            +x, +y, +z,
+
+            +x, +y, +z,
+            +x, -y, -z,
+            +x, +y, -z,
+
+            +x, -y, -z,
+            -x, -y, -z,
+            +x, +y, -z,
+
+            +x, +y, -z,
+            -x, -y, -z,
+            -x, +y, -z,
+
+            -x, -y, -z,
+            -x, -y, +z,
+            -x, +y, -z,
+
+            -x, +y, -z,
+            -x, -y, +z,
+            -x, +y, +z,
+
+            -x, +y, +z,
+            +x, +y, +z,
+            -x, +y, -z,
+
+            -x, +y, -z,
+            +x, +y, +z,
+            +x, +y, -z,
+
+            -x, -y, -z,
+            +x, -y, -z,
+            -x, -y, +z,
+
+            -x, -y, +z,
+            +x, -y, -z,
+            +x, -y, +z,
+        ]);
+    }
+
+    /**
+     * Generates the texture coordinates for the box.
+     */
+    protected generateTexCoords(): Float32Array {
+        return new Float32Array([
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+
+            0.0, 0.0,
+            1.0, 0.0,
+            0.0, 1.0,
+
+            0.0, 1.0,
+            1.0, 0.0,
+            1.0, 1.0,
+        ]);
+    }
+
+    /**
+     * Binds the vertex buffer object (VBO) to an attribute binding point of a given, pre-defined index.
+     */
+    protected bindBuffers(indices: Array<GLuint>): void {
+        /* Please note the implicit bind in attribEnable. */
+        this._buffers[0].attribEnable(0, 3, this.context.gl.FLOAT, false, 0, 0, true, false);
+        if (this._textured) {
+            this._buffers[1].attribEnable(1, 2, this.context.gl.FLOAT, false, 0, 0, true, false);
+        }
+    }
+
+    /**
+     * Unbinds the vertex buffer object (VBO) and disables the binding point.
+     */
+    protected unbindBuffers(indices: Array<GLuint>): void {
+        /* Please note the implicit unbind in attribEnable is skipped. */
+        this._buffers[0].attribDisable(0, true, true);
+        if (this._textured) {
+            this._buffers[1].attribDisable(1, true, true);
+        }
+    }
+
+    /**
+     * Creates the vertex buffer object (VBO) and creates and initializes the buffer's data store.
+     * @param aVertex - Attribute binding point for vertices.
+     */
+    initialize(aVertex: GLuint, aTexCoord: GLuint): boolean {
```

Comment:
please refer to ``NDCfillingtriangle``  - see initialize default attributes + getter vor attribute location queries.